File: src/main/java/pl/arturczopek/mycoach/model/database/Price.java
package pl.arturczopek.mycoach.model.database;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

/**
 * @Author Artur Czopek
 * @Date 09-10-2016
 */

@Data
@Entity
@Table(name = "PRICES")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Price implements Serializable{

    private static final long serialVersionUID = 4215492868990889864L;

    @Id
    @Column(name = "PRC_ID", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "PRICES_PRC_ID_SEQ")
    @SequenceGenerator(name = "PRICES_PRC_ID_SEQ", sequenceName = "PRICES_PRC_ID_SEQ", allocationSize = 1)
    private long priceId;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "PRC_PRD_ID", nullable = false)
    private Product product;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @Column(name = "PRC_DT", nullable = false)
    private Date priceDate;

    @Column(name = "PRC_VAL", nullable = false)
    private float value;

    @Column(name = "PRC_QNT")
    private float quantity;

    @Column(name = "PRC_PLC", nullable = false, length = 60)
    private String place;
}

File: src/main/kotlin/pl/arturczopek/mycoach/model/database/Price.kt
package pl.arturczopek.mycoach.model.database;

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.sql.Date
import javax.persistence.*

/**
 * @Author Artur Czopek
 * @Date 09-10-2016
 */

@Entity
@Table(name = "PRICES")
@JsonIgnoreProperties("hibernateLazyInitializer", "handler")
data class Price(

        @Id
        @Column(name = "PRC_ID", nullable = false)
        @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "PRICES_PRC_ID_SEQ")
        @SequenceGenerator(name = "PRICES_PRC_ID_SEQ", sequenceName = "PRICES_PRC_ID_SEQ", allocationSize = 1)
        var priceId: Long = 0,

        @Column(name = "PRC_PRD_ID", nullable = false)
        var productId: Long = 0,

        @Column(name = "PRC_DT", nullable = false)
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
        var priceDate: Date = Date.valueOf("1970-01-01"),

        @Column(name = "PRC_VAL", nullable = false)
        var value: Float = 0.0f,

        @Column(name = "PRC_QNT")
        var quantity: Float = 0.0f,

        @Column(name = "PRC_PLC", nullable = false, length = 60)
        var place: String = ""
)